errbot/backends/slack.py

```diff
@@ -160,7 +160,8 @@ def __str__(self):
         return self.__unicode__()
 
     def __eq__(self, other):
-        return other.userid == self.userid
+        return other.userid == self.userid and \
+               other.channelid == self.channelid
```

Comment:
I disagree with the semantics here: it is not because the channel you are talking to somebody though is not the same that the person is not the same.